### PR TITLE
Update Travis to GCC 9 and fix warnings

### PR DESCRIPTION
**Pull request by asarium**
Only one warning type this time, the implicit copy operator for types
with a copy constructor has been deprecated so we need to delcare them
ourself.

### CODE REVIEW

File: code/mission/missionparse.cpp [hunk 1]
```diff
@@ -7100,13 +7100,13 @@ void mission_eval_arrivals()
 				if (rship >= 0)
 				{
 					int j;
-					char message_name[NAME_LENGTH + 10];
+					SCP_string message_name;
 					sprintf(message_name, "%s Arrived", wingp->name);
```
Comment (Goober5000) [inline code/mission/missionparse.cpp]: Was this file intended to be part of this PR?  And incidentally, does sprintf support SCP_string?

File: code/weapon/weapons.cpp [hunk 2]
```diff
@@ -2104,9 +2104,9 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		}
 		// now check miss factors for each IFF
 		for(iff=0; iff<Num_iffs; iff++) {
-			char miss_factor_string[NAME_LENGTH + 15];
+			SCP_string miss_factor_string;
 			sprintf(miss_factor_string, "+%s Miss Factor:", Iff_info[iff].iff_name);
```
Comment (Goober5000) [inline code/weapon/weapons.cpp]: Same here

File: code/mission/missionparse.cpp (hunk 1 above)
Comment (asarium) [reply code/mission/missionparse.cpp]: This is intentional. GCC 9 detects more possible buffer overruns when using `sprintf` so I also needed to fix that. `sprintf` support for `SCP_string` is part of FSO and was implemented some time ago.

File: qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp [hunk 3]
```diff
@@ -11,50 +11,43 @@ namespace dialogs {
...
 	AbstractDialogModel(parent, viewport) {
 }
-bool MissionGoalsDialogModel::apply() {
-	char buf[256], names[2][MAX_GOALS][NAME_LENGTH];
-	int i, count;
-
-	for (i=0; i<Num_goals; i++)
+bool MissionGoalsDialogModel::apply()
+{
+	for (int32_t i = 0; i < Num_goals; i++)
```
Comment (Goober5000) [inline qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp]: Why are you using `int32_t` instead of `int` or `auto` or `size_t`?

File: qtfred/src/ui/widgets/sexp_tree.cpp [hunk 4]
```diff
@@ -3945,10 +3945,10 @@ sexp_list_item* sexp_tree::get_listing_opf_ship_wing_shiponteam_point() {
 	sexp_list_item head;
 
 	for (i = 0; i < Num_iffs; i++) {
-		char tmp[NAME_LENGTH + 7];
+		SCP_string tmp;
 		sprintf(tmp, "<any %s>", Iff_info[i].iff_name);
-		strlwr(tmp);
-		head.add_data_dup(tmp);
+		std::transform(begin(tmp), end(tmp), begin(tmp), [](char c) { return (char)::toupper(c); });
```
Comment (Goober5000) [inline qtfred/src/ui/widgets/sexp_tree.cpp]: This transformation uses `toupper`, but the previous one used `strlwr`.

File: qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp (hunk 3 above)
Comment (asarium) [reply qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp]: I am trying to use fixed width integers wherever possible to make the size consistent.

File: qtfred/src/ui/widgets/sexp_tree.cpp (hunk 4 above)
Comment (asarium) [reply qtfred/src/ui/widgets/sexp_tree.cpp]: Indeed, will fix.

File: qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp (hunk 3 above)
Comment (Goober5000) [reply qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp]: But this is a `for` loop. (As are the other instances.) It's not a struct or a memory map. Why should that matter?

Comment (asarium) [reply qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp]: To be honest, that is just the coding standard of my work so I used it out of habit. However, I think it is a worthwhile endeavour to use fixed with types wherever possible so I would prefer to leave it like it is.

File: qtfred/src/ui/widgets/sexp_tree.cpp (hunk 4 above)
Comment (asarium) [reply qtfred/src/ui/widgets/sexp_tree.cpp]: Fixed now.

File: qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp (hunk 3 above)
Comment (Goober5000) [reply qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp]: I wouldn't normally care, but for two reasons:

1) You've been proactive about using `auto` in many other places
2) There have only ever been a maximum of 5 goals.  A three-bit data type would be sufficient.  (The dialog has room for 10 so you could go to a whopping four bits.)  Even a `short` would be suitable.  But you're using a 32-bit type.

Comment (asarium) [reply qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp]: I guess I could use auto instead although that is even longer than `int`...
